test/test_HeadingController.py

```diff
@@ -0,0 +1,354 @@
+
+    # TACKABLE tests
+
+    # Entry to TACKABLE from UNKNOWN
+
+    def test_initialize_tackableMode(self):
+        mock_speed = sailbot_constants.SPEED_THRESHOLD_FOR_JIBING_KNOTS + 0.1
+        mock_time = 0
+
+        hc = HeadingController(mock_speed, mock_time, ControlModes.UNKNOWN.value)
+        self.assertEqual(hc.getControlModeID(), ControlModes.TACKABLE.value)
+
+    # Remain in TACKABLE since exit condition is not met
+
+    def test_noExit_tackableMode(self):
+        mock_speed = sailbot_constants.SPEED_THRESHOLD_FOR_JIBING_KNOTS + 0.1
+        mock_time = 0
+        mock_heading_error = 2 * sailbot_constants.MIN_HEADING_ERROR_FOR_SWITCH
+
+        # Enter TACKABLE mode from UNKNOWN
+        hc = HeadingController(mock_speed, mock_time, ControlModes.UNKNOWN.value)
+
+        # No switch
+        self.assertFalse(hc.switchControlMode(
+            heading_error=mock_heading_error,
+            boat_speed=mock_speed,
+            current_time=mock_time
+        ))
+        self.assertEqual(hc.getControlModeID(), ControlModes.TACKABLE.value)
+
+        # Adjust parameters
+        mock_speed *= 2
+        mock_time += 0.5 * sailbot_constants.SWITCH_INTERVAL
+        mock_heading_error = 1.5 * sailbot_constants.MIN_HEADING_ERROR_FOR_SWITCH
+
+        # No switch
+        self.assertFalse(hc.switchControlMode(
+            heading_error=mock_heading_error,
+            boat_speed=mock_speed,
+            current_time=mock_time
+        ))
+        self.assertEqual(hc.getControlModeID(), ControlModes.TACKABLE.value)
+
+    # Exit TACKABLE due to slow speed
+
+    def test_slowSpeed_exitTackable(self):
+        mock_speed = sailbot_constants.SPEED_THRESHOLD_FOR_JIBING_KNOTS + 0.1
+        mock_time = 0
+
+        # Enter TACKABLE from UNKNOWN
+        hc = HeadingController(mock_speed, mock_time, ControlModes.UNKNOWN.value)
+
+        # Make speed low and adjust time to meet the switch interval
+        mock_speed -= 0.2
+        mock_time += sailbot_constants.SWITCH_INTERVAL
+
+        self.assertTrue(hc.switchControlMode(
+            heading_error=0.5,
+            boat_speed=mock_speed,
+            current_time=mock_time
+        ))
+
+        # Should not be in TACKABLE or UNKNOWN
+        self.assertTrue(hc.getControlModeID() != ControlModes.TACKABLE.value)
+        self.assertTrue(hc.getControlModeID() != ControlModes.UNKNOWN.value)
+
+    # Exit TACKABLE due to reaching the heading
+
+    def test_reachedHeading_exitTackable(self):
+        mock_speed = sailbot_constants.SPEED_THRESHOLD_FOR_JIBING_KNOTS + 0.1
+        mock_time = 0
+
+        # Enter TACKABLE from UNKNOWN
+        hc = HeadingController(mock_speed, mock_time, ControlModes.UNKNOWN.value)
+
+        # Adjust time to meet the switch interval
+        mock_time += sailbot_constants.SWITCH_INTERVAL
+        mock_heading_error = -0.5 * sailbot_constants.MIN_HEADING_ERROR_FOR_SWITCH
+
+        # A switch occurs
+        self.assertTrue(hc.switchControlMode(
```

Comment:
adding a general comment here, do you think it's more clear if we renamed "switchControMode" to something like "reEvaluateControlMode"? it just seems a bit confusing when we assert true on "switchControlMode" and then later assert we are still in the same control mode. 

We can make "reEvaluateControlMode" return true if we actually reevaluated the control mode, which I think is the logic you implemented already